fsm/config.go

```diff
@@ -0,0 +1,118 @@
+package fsm
```

Comment:
I split out the `config` type things from the `fsm.go` file. It was getting large.